A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,548 @@
+|-------------------------------------------|-----------------------------------------|
+| `id`                                      | `message.event.type`                    |
+| `type`                                    | `message.message.id`                    |
+
+And OpenTelemetry no longer has span end options as OpenCensus does.
+
+### Propagator Wire Format
+While gRPC OpenCensus directly interacts with the metadata API, gRPC OpenTelemetry 
+will use the standardized propagators API for context propagation, for the 
+following benefits:
+1. Full integration with OpenTelemetry APIs that is easier for users to reason about.
+2. Make it possible to plugin other propagators that the community supports.
+3. Flexible API that allows clean and simple migration paths to a different propagator.
+
+As of today, OpenTelemetry propagator API only supports `TextMapPropagator`, 
+that is to send string key/value pairs between the client and server, which is 
+different from the binary header that gRPC currently uses. The future roadmap 
+to support binary propagators at OpenTelemetry is unclear. So, gRPC will use 
+propagator API in TextMap format with an optimization path (Go and Java) to work 
+around the lack of binary propagator API to support `grpc-trace-bin`. In fact,
+TextMap propagator does not show visible performance impact for C++, which is 
+the most sensitive language to performance, based on internal micro benchmarking. 
+Therefore, gRPC will only support propagating `grpc-trace-bin` in TextMap propagator.
+
+gRPC will expose a custom `GrpcTraceBinPropagator` that implements `TextMapPropagator`.
+This grpc-provided propagator still uses the `grpc-trace-bin` header for context 
+propagation. When using `grpc-trace-bin` the OpenCensus spanContext and 
+OpenTelemetry spanContext are identical, therefore a gRPC OpenCensus client can 
+speak with a gRPC OpenTelemetry server and vice versa. Users can provide a 
+single composite propagator that combines one or multiple `TextMapPropagator` 
+for their client and server separately. This way, users can define their own 
+migration path for context propagators in distributed components, see detailed
+discussion in the later session. Configuring gRPC OpenTelemetry with this 
+propagator when dealing with cross-process concerns during migration is
+straightforward and recommended. In the long term, community
+standardized propagators, e.g. W3C is more encouraged than `GrpcTraceBinPropagator`. 
+
+####  Propagator API in Java/Go
+The pseudocode below demonstrates `GrpcTraceBinPropagator` and the corresponding 
+gRPC Getter/Setter with an optimization path.
+
+```Java
+public class GrpcTraceBinPropagator implements TextMapPropagator {
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> setter) {
+    SpanContext spanContext = Span.fromContext(context).getSpanContext();
+    byte[] value = BinaryFormat.toBytes(spanContext);
```

Comment:
as discussed offline, the value that we use for "grpc-trace-bin" follows a particular format. We should follow the same format. I've updated the C++ section with the right format. Whether you want to write the entire code is upto you, but you should atleast mention that it will be transformed to the "grpc-trace-bin" format

Reply:
Yea, I verified that with java's binary format implementation (in opencensus repo). Actually in `###Propagator Wire Format` section 3rd paragraph I mentioned similar thing: "When using `grpc-trace-bin` the OpenCensus spanContext and 
OpenTelemetry spanContext are identical, therefore a gRPC OpenCensus client can 
speak with a gRPC OpenTelemetry server and vice versa."